lib/honeybadger/insights/base.ex

```diff
@@ -0,0 +1,175 @@
+defmodule Honeybadger.Insights.Base do
+  @moduledoc """
+  Base module providing common telemetry attachment functionality.
+  """
+
+  defmacro __using__(_opts) do
+    quote do
+      require Logger
+
+      # Define empty defaults for the attributes
+      Module.register_attribute(__MODULE__, :required_dependencies, [])
+      Module.register_attribute(__MODULE__, :telemetry_events, [])
+
+      @before_compile Honeybadger.Insights.Base
+    end
+  end
+
+  defmacro __before_compile__(env) do
+    namespace =
+      env.module
+      |> Atom.to_string()
+      |> String.split(".")
+      |> List.last()
+      |> Macro.underscore()
+      |> String.to_atom()
+
+    quote do
+      # Store the namespace for use in the module. Uses Macro.underscore/1, so
+      # Base.MyModule would become :my_module.
+      @config_namespace unquote(namespace)
+
+      @doc """
+      Checks if all required dependencies are available.
+      """
+      def dependencies_available? do
+        # We are using this strategy to deal with a compiler warning about the
+        # empty case never being reached.
+        deps = @required_dependencies
+
+        if Enum.empty?(deps) do
+          true
+        else
+          Enum.all?(deps, fn mod -> Code.ensure_loaded?(mod) end)
+        end
```

Comment:
`Enum.all?` returns true for an empty list:

```suggestion
        Enum.all?(deps, fn mod -> Code.ensure_loaded?(mod) end)
```

Reply:
Should be addressed in https://github.com/honeybadger-io/honeybadger-elixir/commit/0ad97009fda823273a0d15acb01d9d44951470eb